src/bacnet/basic/tsm/tsm.c

```diff
@@ -19,10 +19,30 @@
 #include "bacnet/datalink/datalink.h"
 #include "bacnet/basic/services.h"
 #include "bacnet/basic/binding/address.h"
+#if BACNET_SEGMENTATION_ENABLED
+#include <stdlib.h>
+#include "bacnet/segmentack.h"
+#include "bacnet/abort.h"
+
+#define DEFAULT_WINDOW_SIZE 32
+
+#ifndef max
+#define max(a, b) (((a) > (b)) ? (a) : (b))
+#endif
+#ifndef min
+#define min(a, b) (((a) < (b)) ? (a) : (b))
+#endif
+
+/*Number of Duplicate Segments Received. */
+static uint8_t Duplicate_Count = 0;
+
+/* Indirection of state machine data with peer unique id values */
+static BACNET_TSM_INDIRECT_DATA TSM_Peer_Ids[MAX_TSM_PEERS];
+#endif // BACNET_SEGMENTATION_ENABLED
 
 /** @file tsm.c  BACnet Transaction State Machine operations  */
 /* FIXME: modify basic service handlers to use TSM rather than this buffer! */
-uint8_t Handler_Transmit_Buffer[MAX_PDU];
+uint8_t *Handler_Transmit_Buffer;
```

Comment:
Either change to `Handler_Transmit_Buffer[MAX_PDU*MAX_SEGMENTS_ACCEPTED];` or add `#if BACNET_SEGMENTATION_ENABLED` around the change and allow both, and place the calloc in the init of this module and not elsewhere.